examples/native-oft-adapter/deploy/MyNativeOFTAdapter.ts

```diff
@@ -0,0 +1,52 @@
+import assert from 'assert'
+
+import { type DeployFunction } from 'hardhat-deploy/types'
+
+const contractName = 'MyNativeOFTAdapter'
+
+const deploy: DeployFunction = async (hre) => {
+    const { getNamedAccounts, deployments } = hre
+
+    const { deploy } = deployments
+    const { deployer } = await getNamedAccounts()
+
+    assert(deployer, 'Missing named deployer account')
+
+    console.log(`Network: ${hre.network.name}`)
+    console.log(`Deployer: ${deployer}`)
+
+    // This is an external deployment pulled in from @layerzerolabs/lz-evm-sdk-v2
+    //
+    // @layerzerolabs/toolbox-hardhat takes care of plugging in the external deployments
+    // from @layerzerolabs packages based on the configuration in your hardhat config
+    //
+    // For this to work correctly, your network config must define an eid property
+    // set to `EndpointId` as defined in @layerzerolabs/lz-definitions
+    //
+    // For example:
+    //
+    // networks: {
+    //   fuji: {
+    //     ...
+    //     eid: EndpointId.AVALANCHE_V2_TESTNET
+    //   }
+    // }
+    const endpointV2Deployment = await hre.deployments.get('EndpointV2')
+
+    const { address } = await deploy(contractName, {
+        from: deployer,
+        args: [
+            18, // TODO update this with native local decimals
```

Comment:
This is where I really dislike deploy scripts, as they do not support mandatory arguments.  `18` is almost always correct, unless it isn't, which is rare but occasionally happens.  EVM has no API to determine gas token decimals that I am aware of (you would likely need to call a known WETH9 precompile).  I think this is the best we can do given the limitations, thoughts?

Reply:
Thanks for the feedback! I agree that hardcoding 18 isn't ideal, but given the lack of an API to determine gas token decimals, this _seems_ like the best option for now. Open to revisiting this if we find a better solution!